spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageProtocolResolver.java

```diff
@@ -84,7 +95,18 @@ public Resource resolve(String location, ResourceLoader resourceLoader) {
 							+ "so defaults are used.");
 				}
 			}
-			return new GoogleStorageResource(this.storage, location, this.settings.isAutoCreateFiles());
+
+			Resource resource;
+
+			if (pathParts.length < 2) {
+				resource = new GoogleStorageResourceBucket(this.storage, bucketName,
+						this.settings.isAutoCreateFiles());
+			}
+			else {
```

Comment:
The local variable `resource` is unnecessary.
You can get rid of it and this else by returning where you have `resource =`

Reply:
Yea I had it like that before, but with the return-null for the gs:/// path above it is 4  return statements in the function and checkstyle disallows that.

Reply:
You might be able to remove the previous `return null` if you throw an exception for invalid path, which I think is actually the better way to go there.
If that doesn't help, feel free to ship like this, though that checkstyle rule has annoyed me before too...

Reply:
That sounds like the way to go. Will change.

Reply:
I would also change the first if to `if (!location.startsWith(PROTOCOL)) { return null; }`, so that the whole method doesn't need to be inside the while loop and doesn't end with that `return null`.